### PR TITLE
WIP - Add retry logic to cluster selection

### PR DESCRIPTION
**Pull request by bfontecc007**
Try a (configurable) number of times to reach a cluster that can handle additional load.

### CODE REVIEW

**Comment by twaugh** [comment]
Firstly, we need to disable the regular retries (using `with cluster_info.osbs.retries_disabled():`, https://github.com/projectatomic/osbs-client/blob/master/osbs/api.py#L1143) for the list-builds and create-build operations. If we don't do that, it will take a long time to move onto the next cluster in the list. We'll have two levels of retrying for the same thing.

Also, we need to apply retry logic to more than just cluster selection. What if creating the build on the selected cluster fails? In that situation we need to try another cluster.

Imagine this situation: clusters A, B, and C are for the same platform, but cluster A is unreachable. What we want to happen then is: try listing builds for cluster A once only -- i.e. discount the cluster for this selection attempt when it fails, then try listing builds for cluster B, then cluster C. After this first pass of attempting to list builds, pick a cluster based on load (e.g. B), then attempt to create the build on it, once only. If creating the build fails, try creating a build on cluster C, once only.

If all those things failed, start again, trying to list builds for cluster A as well as clusters B and C again.

**Comment by bfontecc007** [comment]
Luiz and Vadim, I hadn't seen Tim's comment when I requested your reviews (had to refresh). Looks like I'll probably close this PR and make Tim's changes.

File: tests/plugins/test_orchestrate_build.py [hunk 1]
```diff
@@ -500,6 +500,83 @@ def test_orchestrate_build_choose_clusters(tmpdir, clusters_x86_64,
...
+            }
+        }]
+    )
+
+    runner.run()
+
+
+def test_orchestrate_choose_cluster_retry_timeout(tmpdir):
+
+    (flexmock(OSBS).should_receive('list_builds')
```
Comment (vrutkovs) [inline tests/plugins/test_orchestrate_build.py]: It would be nice to test a case where the results are eventually returned, but after the deadline

**Comment by bfontecc007** [comment]
I have a new branch for this now.

**Pull request closed without merging**